tools/run_tests/sanity/sanity_tests.yaml

```diff
@@ -27,6 +27,7 @@
 - script: tools/distrib/check_trailing_newlines.sh
 - script: tools/distrib/check_upb_output.sh
 - script: tools/distrib/check_pytype.sh
+- script: tools/codegen/core/gen_grpc_tls_credentials_options.py
```

Comment:
This script will generate the new files, but will it fail if the files in the commit do not match the generated files?  I think that's what we need here.

Also, please add the script to generate the new files to tools/distrib/sanitize.sh.

Reply:
Ah, you are right. I though the test checked for file differences on its own. Added a test mode to the script.